Enhance the Error Message When Jenkins Server Is Missing Declarative Jenkins Plugin

If the plugin is missing, you get the below error message (which is nice, thanks for doing that). 
```
[ERROR] Failed to execute goal com.andyserver.maven.plugins:jenkinsfile-maven-plugin:1.0.0:validate (default-cli) on project foo: Jenkinsfile Validation Endpoint
 Not Found. Please Confirm Plugin Installed -> [Help 1]
```

It would be nice to update the text to be a bit more descriptive, that for instance you specifically need [this plugin](https://wiki.jenkins-ci.org/display/JENKINS/Pipeline+Model+Definition+Plugin) on your server.

--- a/src/main/java/com/andyserver/maven/plugins/ValidateJenkinsfileMojo.java
+++ b/src/main/java/com/andyserver/maven/plugins/ValidateJenkinsfileMojo.java
@@ -124,11 +124,11 @@ public class ValidateJenkinsfileMojo extends AbstractMojo {
                 if (statusCode != HttpStatus.SC_OK) {
 
                     if (statusCode == HttpStatus.SC_NOT_FOUND) {
-                        throw new MojoExecutionException(
-                                "Jenkinsfile Validation Endpoint Not Found. Please Confirm Plugin Installed");
+                        throw new MojoExecutionException(String.format(
+                                "%n%nJenkinsfile validation REST endpoint not found on Jenkins server (%s).%nPlease confirm that the pipeline-model-definition plugin is installed.", server));
                     }
 
-                    throw new MojoExecutionException("Invalid HTTP Response: " + statusCode);
+                    throw new MojoExecutionException(String.format("%n%nInvalid HTTP response code: %s.%nPlease note that Jenkins may return a 403 even if your credentials are incorrect, which would normally result in a 401.", statusCode));
                 }
 
                 String response = EntityUtils.toString(httpResponse.getEntity());

--- a/src/main/java/com/andyserver/maven/plugins/ValidateJenkinsfileMojo.java
+++ b/src/main/java/com/andyserver/maven/plugins/ValidateJenkinsfileMojo.java
@@ -128,7 +128,13 @@ public class ValidateJenkinsfileMojo extends AbstractMojo {
                                 "%n%nJenkinsfile validation REST endpoint not found on Jenkins server (%s).%nPlease confirm that the pipeline-model-definition plugin is installed.", server));
                     }
 
-                    throw new MojoExecutionException(String.format("%n%nInvalid HTTP response code: %s.%nPlease note that Jenkins may return a 403 even if your credentials are incorrect, which would normally result in a 401.", statusCode));
+                    String errorMessageTemplate = "%n%nInvalid HTTP response code: %s.";
+
+                    if (statusCode == HttpStatus.SC_FORBIDDEN){
+                        errorMessageTemplate += "%nPlease note that Jenkins may return a 403 even if your credentials are incorrect, which would normally result in a 401.";
+                    }
+
+                    throw new MojoExecutionException(String.format(errorMessageTemplate, statusCode));
                 }
 
                 String response = EntityUtils.toString(httpResponse.getEntity());

